ext/imap/php_imap.c

```diff
@@ -1245,28 +1266,33 @@ PHP_FUNCTION(imap_body)
 		RETURN_THROWS();
 	}
 
-	if (flags && ((flags & ~(FT_UID|FT_PEEK|FT_INTERNAL)) != 0)) {
-		php_error_docref(NULL, E_WARNING, "Invalid value for the options parameter");
-		RETURN_FALSE;
+	if ((imap_le_struct = (pils *)zend_fetch_resource(Z_RES_P(streamind), "imap", le_imap)) == NULL) {
+		RETURN_THROWS();
 	}
 
-	if ((imap_le_struct = (pils *)zend_fetch_resource(Z_RES_P(streamind), "imap", le_imap)) == NULL) {
+	if (msgno < 1) {
+		zend_argument_value_error(2, "must be greater than 0");
 		RETURN_THROWS();
 	}
 
-	if ((argc == 3) && (flags & FT_UID)) {
+	if (flags && ((flags & ~(FT_UID|FT_PEEK|FT_INTERNAL)) != 0)) {
+		zend_argument_value_error(3, "must be a bitmask of FT_UID, FT_PEEK, and FT_INTERNAL");
+		RETURN_THROWS();
+	}
+
+	if (flags && (flags & FT_UID)) {
 		/* This should be cached; if it causes an extra RTT to the
 		   IMAP server, then that's the price we pay for making
 		   sure we don't crash. */
 		msgindex = mail_msgno(imap_le_struct->imap_stream, msgno);
 	} else {
 		msgindex = msgno;
 	}
-	if ((msgindex < 1) || ((unsigned) msgindex > imap_le_struct->imap_stream->nmsgs)) {
-		php_error_docref(NULL, E_WARNING, "Bad message number");
-		RETURN_FALSE;
-	}
 
+	// TODO int overflow as can be seen in imap_body.phpt will result in 999 => <0 trigerring a ValueError
```

Comment:
I don't get this comment. How can 999 overflow?

Reply:
Well, I don't understand how I trigger the ValueError for less than 1, I need to look into it more anywats